includes/class-wcpdf-frontend.php

```diff
@@ -146,25 +145,9 @@ public function download_invoice_shortcode( $atts ) {
 			$link_text = __( 'Download invoice (PDF)', 'woocommerce-pdf-invoices-packing-slips' );
 		}
 
-		// Basic query args for PDF	
-		$query_args = array(
-			'action'        => 'generate_wpo_wcpdf',
-			'document_type' => 'invoice',
-			'order_ids'     => $order->get_id(),
-		);
-		
-		// Add query args based on user permissions and guest access setting
-		$debug_settings = get_option( 'wpo_wcpdf_settings_debug', array() );
-		if( is_user_logged_in() ) {
-			$query_args['my-account'] = true;
-		} elseif( ! is_user_logged_in() && isset( $debug_settings['guest_access'] ) ) {
-			$query_args['order_key'] = $order->get_order_key();
-		} else {
-			return; // no business here
-		}
-	
-		$pdf_url = esc_url( wp_nonce_url( add_query_arg( $query_args, admin_url( 'admin-ajax.php' ) ), 'generate_wpo_wcpdf' ) );
-		$text = sprintf( '<p><a href="%s" target="_blank">%s</a></p>', esc_attr( $pdf_url ), esc_html( $link_text ) );
+		$pdf_url = WPO_WCPDF()->endpoint->get_document_link( $order, 'invoice', array( 'shortcode' => 'true' ) );
```

Comment:
What is the `shortcode` query arg for? I don't think this is used? Keep in mind that this is a customer facing URL.

Reply:
I add it to identify when using with shortcode, I can remove if you prefer.

Reply:
where are you doing that identification? and for what purpose?

Reply:
Was if just to identify like we do with `my-account`. Can be used in the future, but is not being used right now.

Reply:
for `my-account` we actually do use the variable in the checks (although I think it's not absolutely necessary). If you're not using it now, I suggest removing it.